src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/DESCryptoServiceProvider.cs

```diff
@@ -0,0 +1,85 @@
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using Internal.Cryptography;
+using Internal.NativeCrypto;
+
+namespace System.Security.Cryptography
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public sealed class DESCryptoServiceProvider : DES
+    {
+        private const int BitsPerByte = 8;
+        private static readonly RandomNumberGenerator s_rng = RandomNumberGenerator.Create();
+
+        public DESCryptoServiceProvider() { }
+
+        public override void GenerateKey()
+        {
+            var key = new byte[8];
+            s_rng.GetBytes(key);
+            // Never hand back a weak or semi-weak key
+            while (IsWeakKey(key) || IsSemiWeakKey(key))
+            {
+                s_rng.GetBytes(key);
+            }
+            KeyValue = key;
+        }
+
+        public override void GenerateIV()
+        {
+            var iv = new byte[8];
+            s_rng.GetBytes(iv);
+            IVValue = iv;
+        }
+
+        public override ICryptoTransform CreateDecryptor()
+        {
+            return CreateTransform(this.Key, this.IV, encrypting: false);
+        }
+
+        public override ICryptoTransform CreateDecryptor(byte[] rgbKey, byte[] rgbIV)
+        {
+            return CreateTransform(rgbKey, rgbIV.CloneByteArray(), encrypting: false);
+        }
+
+        public override ICryptoTransform CreateEncryptor()
+        {
+            return CreateTransform(this.Key, this.IV, encrypting: true);
+        }
+
+        public override ICryptoTransform CreateEncryptor(byte[] rgbKey, byte[] rgbIV)
+        {
+            return CreateTransform(rgbKey, rgbIV.CloneByteArray(), encrypting: true);
+        }
+
+        private ICryptoTransform CreateTransform(byte[] rgbKey, byte[] rgbIV, bool encrypting)
+        {
+            // note: rgbIV is guaranteed to be cloned before this method, so no need to clone it again
+
+            if (rgbKey == null)
+                throw new ArgumentNullException(nameof(rgbKey));
+
+            long keySize = rgbKey.Length * (long)BitsPerByte;
+            if (keySize > int.MaxValue || !((int)keySize).IsLegalSize(this.LegalKeySizes))
+                throw new ArgumentException(SR.Cryptography_InvalidKeySize, nameof(rgbKey));
+
+            if (IsWeakKey(rgbKey))
+                throw new CryptographicException(SR.Cryptography_InvalidKey_Weak, "DES");
+            if (IsSemiWeakKey(rgbKey))
+                throw new CryptographicException(SR.Cryptography_InvalidKey_SemiWeak, "DES");
+
+            // We truncate IV's that are longer than the block size to 8 bytes : this is
+            // done to maintain backward desktop compatibility with the behavior shipped in V1.x.
+            // The call to set the IV in CryptoAPI will ignore any bytes after the first 8
+            // bytes. We'll still reject IV's that are shorter than the block size though.
+            if (rgbIV.Length < 8)
```

Comment:
```
DES des = new DESCryptoServiceProvider();
des.Mode = CipherMode.ECB;
ICryptoTransform encryptor = des.CreateEncryptor(key, null);
```

Since the CloneByteArray extension method returns null for a null input, that'll nullref here.